File: src/Java/217.ContainsDuplicate.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.Map;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> s = new HashSet<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            if (s.contains(nums[i]) == true) {
                return true;
            } else {
                s.add(nums[i]);
            }
        }
        return false;
    }
}

File: src/Python/217.ContainsDuplicate.py

class Solution:
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        n = len(nums)
        if n == 0:
            return False
        s = {nums[0]}
        for i in range(1, n):
            if nums[i] in s:
                return True
            else:
                s.add(nums[i])
        return False

if __name__ == '__main__':
    solution = Solution()
    print(solution.containsDuplicate([1,2,2,3])
else:
    pass
